keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/unary/PermuteVertexTest.java

```diff
@@ -0,0 +1,202 @@
+package io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.unary;
+
+import com.google.common.collect.ImmutableSet;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.dbl.Differentiator;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.binary.MultiplicationVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.UniformVertex;
+import org.junit.Assert;
+import org.junit.Test;
+
+import static io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.unary.UnaryOperationTestHelpers.testWithFiniteDifference;
+
+public class PermuteVertexTest {
+
+    @Test
+    public void canPermuteForTranpose() {
+        DoubleVertex a = new UniformVertex(0, 10);
+        a.setValue(DoubleTensor.create(new double[]{1, 2, 3, 4, 5, 6}, 2, 3));
+
+        PermuteVertex transpose = new PermuteVertex(a, 1, 0);
+
+        Assert.assertArrayEquals(new long[]{3, 2}, transpose.getShape());
+        Assert.assertArrayEquals(a.getValue().transpose().asFlatDoubleArray(), transpose.getValue().asFlatDoubleArray(), 1e-6);
+    }
+
+    @Test
+    public void changesMatchGradientRankTwo() {
+        testWithFiniteDifference(v -> v.permute(1, 0), new long[]{4, 5});
+        testWithFiniteDifference(v -> v.permute(0, 1), new long[]{4, 5});
+    }
+
+    @Test
+    public void changesMatchGradientRankThree() {
+        testWithFiniteDifference(v -> v.permute(1, 2, 0), new long[]{4, 3, 2});
+        testWithFiniteDifference(v -> v.permute(0, 1, 2), new long[]{4, 3, 2});
+        testWithFiniteDifference(v -> v.permute(2, 1, 0), new long[]{4, 3, 2});
+    }
+
+    @Test
+    public void canCalculateAutoDiffOfRankThreePermute() {
```

Comment:
A lot of these tests aren't testing much more than what the finite difference tests are testing. What more they're testing is really more of a Differentiator or other operation (e.g. times, sum) concern.